docs/scenarios_authoring/RRDataSourcesAndSinks.md

```diff
@@ -2,42 +2,73 @@
 
-- Input schema
-- Output schema
+### Is Nussknacker request response scenario a decision tree?
+The way in which the scenario diagram should be interpreted in the Request - Response processing mode requires some attention. The request - response scenario diagram is not a [decision tree](https://en.wikipedia.org/wiki/Decision_tree). During a decision tree "processing" one and only one "branch" of a tree is "activated"; "processing" ends up in one of the end leaves. As in a decision tree each leaf represents a decision (or result) - one decision (result) is generated.
 
-Nussknacker uses those schema to detect how the request and response should look like.
+In Request-Response processing mode, the diagram is more like a flow chart. The data record processed by the scenario (#input at the very start of the processing) "flows" through the scenario. If there are [splits](./BasicNodes.md#split), the data records start to "flow" in parallel through many branches. If there are [for-each](./BasicNodes.md#foreach) nodes, multiple records are produced as the result. Because the Request-Response scenario can return only one response, a question arises what exactly will be returned in such cases and how to - if neccessary - return all data records produced in a scenario. 
 
-![RR schema](img/rrProperties.png "RR properties")
+### Scenario response in scenarios with split and for-each nodes
+If parallel branches end with response nodes and more than one response is generated (e.g there is no filtering after split), the only response which is returned is the one which was chronologically generated first. If no response is generated (e.g. after filtering), an error is returned (in the future this behaviour may change).
+
+If you want to to return data from multiple parallel branches, use [union](./BasicNodes.md#union) node to merge branches and [collect](./BasicNodes.md#collect) node to collect data records into a list. Your scenario should have only one response node.
+
+If you use for-each node and need to collect all the results into a list of values, use [collect](#collect) node.
 
-## Request-Response source
+## Components
+### Common configuration - Request-Response schema
+
+See [schemas](#schemas) subsection of the Concepts section for the rationale behind schemas. 
+
+Click Properties icon on the right panel to define input and output schemas.
+
+![RR schema](img/rrProperties.png "RR properties")
 
-Thanks to the [Input schema](#request-response-schema) Nussknacker knows what data to expect in the request, so it will
-hint field names and will perform additional validations on SpEL expressions involving data from the selected topic.
+### Request-Response source
 
-![RR source](img/rrSource.png "RR source")
+The request node is very simple - it does not require any configuration. Just drop it on the diagram and proceed with authoring the scenario.
```

Comment:
It is, but maybe we add something about input schema and how we can later use defined in schema properties in scenario? ex. `#input.name` etc.

Reply:
The section before - "one sentence away" is: _Click Properties icon on the right panel to define input and output schemas_

Reply:
Hmm, you are right - it should be enough :)